app/src/main/java/com/monkeyteam/chimpagne/ui/event/polls/PollAndVotingScreen.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.monkeyteam.chimpagne.R
+import com.monkeyteam.chimpagne.model.database.ChimpagneRole
+import com.monkeyteam.chimpagne.ui.components.ChimpagneButton
+import com.monkeyteam.chimpagne.ui.components.Legend
+import com.monkeyteam.chimpagne.viewmodels.AccountViewModel
+import com.monkeyteam.chimpagne.viewmodels.EventViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PollsAndVotingScreen(
+    eventViewModel: EventViewModel,
+    accountViewModel: AccountViewModel,
+    onGoBack: () -> Unit
+) {
+  val eventUIState by eventViewModel.uiState.collectAsState()
+  val accountUIState by accountViewModel.uiState.collectAsState()
+
+  var displayCreatePollPopup by remember { mutableStateOf(false) }
+  var displayVotePollPopup by remember { mutableStateOf(false) }
+  var displayViewPollPopup by remember { mutableStateOf(false) }
+  var selectedPollId by remember { mutableStateOf("") }
+
+  if (displayCreatePollPopup) {
+    CreatePollDialog(
+        onPollCreate = {
+          eventViewModel.createPollAtomically(
+              poll = it, onSuccess = { displayCreatePollPopup = false })
+        },
+        onPollCancel = { displayCreatePollPopup = false },
+        onDismissRequest = { displayCreatePollPopup = false })
+  }
+  if (displayVotePollPopup) {
+    val poll = eventUIState.polls[selectedPollId]!!
+    VotePollDialog(
+        poll = poll,
+        userRole = eventUIState.currentUserRole,
+        onOptionVote = {
+          eventViewModel.castPollVoteAtomically(
+              pollId = selectedPollId, optionIndex = it, { displayViewPollPopup = true })
+        },
+        onPollDelete = {
+          eventViewModel.deletePollAtomically(
+              pollId = it, onSuccess = { displayVotePollPopup = false })
+        },
+        onPollCancel = { displayVotePollPopup = false },
+        onDismissRequest = { displayVotePollPopup = false })
+  }
+  if (displayViewPollPopup) {
+    displayVotePollPopup = false
+    val poll = eventUIState.polls[selectedPollId]!!
+    ViewPollDialog(
+        poll = poll,
+        selectedOptionId = poll.votes[accountUIState.currentUserUID]!!,
+        userRole = eventUIState.currentUserRole,
+        onPollDelete = {
+          eventViewModel.deletePollAtomically(
+              pollId = it, onSuccess = { displayViewPollPopup = false })
+        },
+        onPollCancel = { displayViewPollPopup = false },
+        onDismissRequest = { displayViewPollPopup = false })
+  }
+  Scaffold(
+      topBar = {
+        TopAppBar(

```

Comment:
please use `com.monkeyteam.chimpagne.ui.components.TopBar`

Reply:
You already did this in another PR. So it has been fixed.